pcsx2/Gif.cpp

```diff
@@ -194,12 +311,15 @@ static __fi tDMA_TAG* ReadTag2()
 
 bool CheckPaths(EE_EventType Channel) {
 	// Can't do Path 3, so try dma again later...
-	if(!gifUnit.CanDoPath3()) {
-		if(!gifUnit.Path3Masked())
-		{
-			CPU_INT(Channel, 128);
+	if (!CHECK_GIFFIFOHACK) {
+		if (!gifUnit.CanDoPath3()) {
+			if (!gifUnit.Path3Masked())
+			{
```

Comment:
Yes if the hack is enabled we do just return true (as in carry on, this is handled in the fifo) but if it's not enabled we revert to the old functionality.